test/extensions/filters/http/jwt_authn/authenticator_test.cc

```diff
@@ -251,6 +251,179 @@ TEST_F(AuthenticatorTest, TestSetHeader) {
+  // Test JwtExpired pubkey and its cache
+  Http::TestRequestHeaderMapImpl headers{{"Authorization", "Bearer " + std::string(ExpiredToken)}};
+
+  expectVerifyStatus(Status::JwtExpired, headers);
+
+  // Only one field is set.
+  EXPECT_EQ(1, out_extracted_data_.fields().size());
+
+  EXPECT_EQ(enumToInt(Status::JwtExpired), out_extracted_data_.fields()
+                                               .at("jwt-failure-reason")
+                                               .struct_value()
+                                               .fields()
+                                               .find("code")
+                                               ->second.number_value());
+
+  EXPECT_EQ(google::jwt_verify::getStatusString(Status::JwtExpired), out_extracted_data_.fields()
+                                                                         .at("jwt-failure-reason")
+                                                                         .struct_value()
+                                                                         .fields()
+                                                                         .find("message")
+                                                                         ->second.string_value());
+}
+
+// This test verifies jwt status in failed status in metadata with allow missing or failed
+TEST_F(AuthenticatorTest, TestSetInvalidJwtToGetStatus) {
+  // Config provider and failed status in metadata
+  auto& provider = (*proto_config_.mutable_providers())[std::string(ProviderName)];
+  provider.set_failed_status_in_metadata("jwt-failure-reason");
+  createAuthenticator(nullptr, absl::make_optional<std::string>(ProviderName),
+                      /*allow_failed=*/true, /*allow_missing=*/true);
+  EXPECT_CALL(*raw_fetcher_, fetch(_, _)).Times(0);
+
+  // Test JwtAudienceNotAllowed pubkey and its cache
+  Http::TestRequestHeaderMapImpl headers{
+      {"Authorization", "Bearer " + std::string(InvalidAudToken)}};
+
+  expectVerifyStatus(Status::Ok, headers);
+
+  // Only one field is set.
+  EXPECT_EQ(1, out_extracted_data_.fields().size());
+
+  EXPECT_EQ(enumToInt(Status::JwtAudienceNotAllowed), out_extracted_data_.fields()
+                                                          .at("jwt-failure-reason")
+                                                          .struct_value()
+                                                          .fields()
+                                                          .find("code")
+                                                          ->second.number_value());
+
+  EXPECT_EQ(google::jwt_verify::getStatusString(Status::JwtAudienceNotAllowed),
+            out_extracted_data_.fields()
+                .at("jwt-failure-reason")
+                .struct_value()
+                .fields()
+                .find("message")
+                ->second.string_value());
+}
+
+// This test verifies jwt missing status in failed status in metadata with allow missing or failed
```

Comment:
// This test verifies writing JwtMissing status into metadata
// with allow_failed=true and allow_mssing=true
// with single provider

Please add another test case with two providers, and verify that we did not write to metadata.